libs/langchain-community/src/embeddings/alibaba_embedding.ts

```diff
@@ -0,0 +1,215 @@
+import { getEnvironmentVariable } from "@langchain/core/utils/env"
+import { Embeddings, type EmbeddingsParams } from "@langchain/core/embeddings"
+import { chunkArray } from "@langchain/core/utils/chunk_array"
+
+export interface AlibabaAIEmbeddingsParams extends EmbeddingsParams {
+    /** Model name to use */
+    // modelName: "text-embedding-async-v2" | "text-embedding-v2"
+    modelName: "text-embedding-v2"
+
+    /**
+     * The number of dimensions the resulting output embeddings should have.
+     * Only supported in `text-embedding-3` and later models.
+     */
+    // dimensions?: number
+
+    /**
+     * Timeout to use when making requests to OpenAI.
+     */
+    timeout?: number
+
+    /**
+     * The maximum number of documents to embed in a single request. This is
+     * limited by the OpenAI API to a maximum of 2048.
+     */
+    batchSize?: number
+
+    /**
+     * Whether to strip new lines from the input text. This is recommended by
+     * OpenAI for older models, but may not be suitable for all use cases.
+     * See: https://github.com/openai/openai-python/issues/418#issuecomment-1525939500
+     */
+    stripNewLines?: boolean
+
+    parameters?: {
+        /**
+         * 取值：query 或者 document，默认值为 document
+         * 说明：文本转换为向量后可以应用于检索、聚类、分类等下游任务，
+         * 	对检索这类非对称任务为了达到更好的检索效果建议区分查询文本（query）和
+         * 	底库文本（document）类型, 聚类、分类等对称任务可以不用特殊指定，
+         * 	采用系统默认值"document"即可
+         */
+        text_type?: "text" | "document"
+    }
+}
+
+interface EmbeddingCreateParams {
+    model: AlibabaAIEmbeddingsParams["modelName"]
+    input: {
+        texts: string[]
+    }
+
+    parameters?: AlibabaAIEmbeddingsParams["parameters"]
+}
+
+interface EmbeddingResponse {
+    output: {
+        embeddings: { text_index: number; embedding: number[] }[]
+    }
+    usage: {
+        total_tokens: number
+    }
+    request_id: string
+}
+
+interface EmbeddingErrorResponse {
+    code: string
+    message: string
+    request_id: string
+}
+
+export class AlibabaAIEmbeddings extends Embeddings implements AlibabaAIEmbeddingsParams {
```

Comment:
See above, rename to `AlibabaTongyiEmbeddingsParams`

Reply:
I changed it